### PR TITLE
Regression User field is not validating

### LINKED ISSUE #8667: REGRESSION user notes userid
Testing in current staging when creating a user note the field for the User is marked as required but you can still save a note with selecting a user - making the whole thing useless

Previously it was not possible to save a user note without completing the required field

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### This fixes #8667

Field is hardcoded as disabled and the required option is absent

Also contains some code style fixes for the fields: user, media, contenthistory (the var block comments are unified)
#### Testing

Create a new User Note
do not select user
Form should not be valid and not therefore not been submitted (but it is)

Apply patch and re do the process

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on a8b0f3fdd88b482842afeaffca4d0f14846cb812<br /><br />Didnt work

Applied patch
Selected Add New User Note
Clicked Save
User Note was saved<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9037">issues.joomla.org/joomla-cms/9037</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 95de2621abe7bb32c986fe11d9f9bd47caed8ef3<br /><br />Almost works - cant be saved but no error displayed - just an empty error div<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9037">issues.joomla.org/joomla-cms/9037</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 99a3ae1f2ec22644c2fc511c6a4899d0141cb074<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9037">issues.joomla.org/joomla-cms/9037</a>.</sub>

**Comment by infograf768** [comment]
Works now here too.

Can you explain further (maybe with comments):
```
+if (JText::_('JLIB_FORM_SELECT_USER') == htmlspecialchars($userName, ENT_COMPAT, 'UTF-8'))
+{
+	$userName = "";
+}
+
```

**Comment by dgt41** [comment]
@infograf768 is a simple check to invalidate the value of the input. If $userName is "Select User" then we set the value to null, so validation will fail and we will get a message. This only happens in the first time the field is rendered, and a user is not already selected.

**Comment by brianteeman** [comment]
Setting RTC - we have two good tests - thanks @dgt41 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9037">issues.joomla.org/joomla-cms/9037</a>.</sub>

**Comment by dgt41** [comment]
For completeness: 
If the field is set as required then a user should not be able to unset it. In other words the button no user in the modal should appear only if the field can have no value!

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 122923c63f1fad0ff67ac0453ff18633d16ea4fa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9037">issues.joomla.org/joomla-cms/9037</a>.</sub>

**Comment by infograf768** [comment]
This should go into 3.5.0

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 122923c63f1fad0ff67ac0453ff18633d16ea4fa<br /><br />Tested with success as described on current staging + this patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9037">issues.joomla.org/joomla-cms/9037</a>.</sub>

File: administrator/components/com_users/views/users/tmpl/modal.php [hunk 1]
```diff
@@ -30,8 +30,10 @@
 			<div class="btn-group pull-left">
 				<button type="submit" class="btn hasTooltip" title="<?php echo JHtml::tooltipText('JSEARCH_FILTER_SUBMIT'); ?>" data-placement="bottom"><span class="icon-search"></span></button>
 				<button type="button" class="btn hasTooltip" title="<?php echo JHtml::tooltipText('JSEARCH_FILTER_CLEAR'); ?>" data-placement="bottom" onclick="document.getElementById('filter_search').value='';this.form.submit();"><span class="icon-remove"></span></button>
-				<button type="button" class="btn button-select" data-user-value="0" data-user-name="<?php echo $this->escape(JText::_('JLIB_FORM_SELECT_USER')); ?>"
-					data-user-field="<?php echo $this->escape($field);?>"><?php echo JText::_('JOPTION_NO_USER'); ?></button>
+				<?php if ($input->get('required', 0, 'INT') != 1 ) : ?>
```
Comment (roland-d) [inline administrator/components/com_users/views/users/tmpl/modal.php]: @dgt41 Can you make the INT lowercase please? Thanks.

**Comment by Kubik-Rubik** [comment]
@dgt41 Thank you!

@roland-d Sorry, did not see your note. @dgt41 Could you make the change in a new PR? Thanks!

**Comment by joomdonation** [comment]
Maybe we can just leave it as how it is:

1. Uppercase filter string is used in other places in Joomla core, for example https://github.com/joomla/joomla-cms/blob/staging/components/com_users/controllers/user.php#L38-L42

2. The docblock of JFilterInput clean methods seems suggest that we use uppper case https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/filter/input.php#L206-L220

**Comment by dgt41** [comment]
@Kubik-Rubik check #9047

**Pull request merged**